### ISSUE TITLE
[v1.0.0] TypeError: list indices must be integers or slices, not str

### ISSUE THREAD

**Issue by GuidoBartoli** [dmlc/xgboost#5353]
Hi,
We are using **XGBoost 0.9** with a custom script that load datasets from HDF5 and train or test an XGBClassifier with the ability to tune parameters and evaluate results. We are using it in production and we did not have any issue until now.
When we upgraded to **v1.0.0**, the same script gives this error when `fit()` function is called:

```
Traceback (most recent call last):
  File "iris.py", line 12, in <module>
    model.fit(train_x, train_y, verbose=True, eval_set=[(train_x, train_y), (test_x, test_y)], early_stopping_rounds=20, eval_metric=['merror', 'mlogloss'])
  File "/home/bartoli/.virtualenvs/xgb/lib/python3.6/site-packages/xgboost/sklearn.py", line 823, in fit
    callbacks=callbacks)
  File "/home/bartoli/.virtualenvs/xgb/lib/python3.6/site-packages/xgboost/training.py", line 209, in train
    xgb_model=xgb_model, callbacks=callbacks)
  File "/home/bartoli/.virtualenvs/xgb/lib/python3.6/site-packages/xgboost/training.py", line 42, in _train_internal
    'num_parallel_tree'] is not None:
TypeError: list indices must be integers or slices, not str
```

It seems that the error is raised by XGBoost when we use two metrics in `eval_metric` parameters in the `fit()` function as we did before. We wrote a small script to test this behaviour:

```
import xgboost as xgb
from h5py import File
from sklearn.model_selection import train_test_split

db = File('iris.h5', 'r')
train_x, test_x, train_y, test_y = train_test_split(db['data'][:], db['target'][:], test_size=0.2)
model = xgb.XGBClassifier(objective='multi:softmax', n_jobs=-1, n_estimators=100,
                          max_depth=5, learning_rate=0.1, subsample=0.8, colsample_bytree=0.9)

# This call works
model.fit(train_x, train_y, verbose=True, eval_set=[(train_x, train_y), (test_x, test_y)], early_stopping_rounds=20)

# This call does not work
model.fit(train_x, train_y, verbose=True, eval_set=[(train_x, train_y), (test_x, test_y)], early_stopping_rounds=20, eval_metric=['merror', 'mlogloss'])
```

This is the example dataset to test it out, can you reproduce this bug?

[iris.zip](https://github.com/dmlc/xgboost/files/4244097/iris.zip)

**Comment by trivialfis** [comment]
Yes.  It's fixed in 1.0.2 pending as a patch release: https://github.com/dmlc/xgboost/issues/5342

**Comment by hcho3** [comment]
@GuidoBartoli We just release 1.0.2.

**Comment by GuidoBartoli** [comment]
> @GuidoBartoli We just released 1.0.2.

That’s really a good news, I’ll update and try again, thanks!
